### PR TITLE
Add various modules

### PR DESCRIPTION
**Pull request by MancunianSam**
* Cloudwatch events
* IAM Policy
* IAM ROle
* SNS
* SQS
* SSM Parameter
* VPC

### CODE REVIEW

File: sns/main.tf [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+  protocol  = "lambda"
+  topic_arn = aws_sns_topic.sns_topic.arn
+}
+
+resource "aws_sns_topic_subscription" "sqs_subscriptions" {
+  for_each  = var.sqs_subscriptions
+  endpoint  = each.value
+  protocol  = "sqs"
+  topic_arn = aws_sns_topic.sns_topic.arn
+}
```
[the remark this answers is not in the document]
Comment (MancunianSam) [reply sns/main.tf:26]: I can't see why we wouldn't. I've switched it on for both topic subscriptions. We can always add a parameter if someone desperately wants to turn if off later.

File: sqs/main.tf [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+    tomap(
+      { Name = var.queue_name }
+    )
+  )
+  visibility_timeout_seconds = var.visibility_timeout
+}
+
+resource "aws_sqs_queue" "dlq" {
+  name = "${var.queue_name}-dlq"
+}
```
[the remark this answers is not in the document]
Comment (MancunianSam) [reply sqs/main.tf:26]: Good point, I've added this in as a static value.

File: vpc/main.tf [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+resource "aws_subnet" "private" {
+  count             = var.az_count
+  cidr_block        = cidrsubnet(aws_vpc.main.cidr_block, var.new_bits, count.index)
+  availability_zone = data.aws_availability_zones.available.names[count.index]
+  vpc_id            = aws_vpc.main.id
+
+  tags = merge(
+    var.tags,
+    tomap(
+      { "Name" = "${var.vpc_name}-private-subnet-${count.index}" }
```
[the remark this answers is not in the document]
Comment (MancunianSam) [reply vpc/main.tf:28]: Yeah makes sense, I've changed it.

File: vpc/variables.tf [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+}
+
+variable "cidr_block" {
+  default = "10.0.0.0/16"
+}
+
+variable "new_bits" {
+  default     = 8
+  description = "The number of bits to extend the network prefix by when creating subnets. A /16 VPC CIDR block and a new_bits value of 8 will return a /24 subnet"
+}
```
[the remark this answers is not in the document]
Comment (MancunianSam) [reply vpc/variables.tf:22]: Terraform calls it newbits but it is a bit weird. I've changed it so you can put `/24` or `/20`. It's a tiny bit more string fiddling but it's notation people are familiar with.

**Comment by MancunianSam** [comment]
I've added a security group module as well since this was last checked.

File: lambda/main.tf [hunk 5]
```diff
@@ -85,7 +85,7 @@ resource "aws_lambda_permission" "lambda_permissions" {
 
 resource "aws_iam_role" "lambda_iam_role" {
   assume_role_policy = templatefile("${path.module}/templates/lambda_assume_role.json.tpl", {})
-  name               = "${var.function_name}Role"
+  name               = "${var.function_name}-role"
```
Comment (kurtismash) [inline lambda/main.tf:88]: Does this need `role` appended? We know it's a role as it's in IAM and the ARN will contain `:role/...`

**Review by kurtismash — APPROVED** [review]

**Pull request merged**